pkg/providers/aws/provider_redis_metrics_test.go

```diff
@@ -0,0 +1,248 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/aws/aws-sdk-go/service/cloudwatch/cloudwatchiface"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/cloud-resource-operator/pkg/moq/moq_aws"
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers"
+	"github.com/sirupsen/logrus"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+const (
+	testRedisMetricName  = "mock_result_id"
+	testRedisMetricValue = 1.11111
+)
+
+func buildReplicationGroupReadyCacheClusterId() []*elasticache.ReplicationGroup {
+	var (
+		cacheClusterId1 = "test-001"
+		cacheClusterId2 = "test-002"
+	)
+	return []*elasticache.ReplicationGroup{
+		{
+			ReplicationGroupId:     aws.String("testtesttest"),
+			Status:                 aws.String("available"),
+			CacheNodeType:          aws.String("test"),
+			SnapshotRetentionLimit: aws.Int64(20),
+			MemberClusters:         []*string{&cacheClusterId1, &cacheClusterId2},
+			NodeGroups: []*elasticache.NodeGroup{
+				{
+					NodeGroupId:      aws.String("primary-node"),
+					NodeGroupMembers: nil,
+					PrimaryEndpoint: &elasticache.Endpoint{
+						Address: testAddress,
+						Port:    testPort,
+					},
+					Status: aws.String("available"),
+				},
+			},
+		},
+	}
+}
+
+var testRedisMetricLabels1 = map[string]string{
+	"clusterID":   "test",
+	"instanceID":  "test-001",
+	"namespace":   "test",
+	"productName": "",
+	"resourceID":  "test",
+	"strategy":    "aws-elasticache",
+}
+var testRedisMetricLabels2 = map[string]string{
+	"clusterID":   "test",
+	"instanceID":  "test-002",
+	"namespace":   "test",
+	"productName": "",
+	"resourceID":  "test",
+	"strategy":    "aws-elasticache",
+}
```

Comment:
Would it be worth adding these to a single build mock labels function, so that we can easily override them inline in the test :thinking: